### PR TITLE
HDDS-6683. Refactor OM server bucket layout configuration usage

### PR DESCRIPTION
**Pull request by tanvipenumudy**
## What changes were proposed in this pull request?

After the post finalization, user should be able to create LEGACY, FSO, OBS layouts.
 
The proposal is to modify the behavior on the server-side to only use the ozone.default.bucket.layout config if the client has not specified any bucket layout in the protobuf message.

If the client sends LEGACY bucket layout type then OM server will not override it with ozone.default.bucket.layout config value. This way a client has the flexibility to create a LEGACY bucket explicitly.

Hint: We can use the `bucketInfo.hasBucketLayout()` Protobuf method to check if the field is set by the user.

Reference: [Discussion here](https://github.com/apache/ozone/pull/3411#issuecomment-1126550908)

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6683

## How was this patch tested?

Added unit and integration tests.

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketCreateRequest.java [hunk 1]
```diff
@@ -152,8 +151,12 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
     OMResponse.Builder omResponse = OmResponseUtil.getOMResponseBuilder(
         getOmRequest());
     OmBucketInfo omBucketInfo = null;
-    if (bucketInfo.getBucketLayout() == null || bucketInfo.getBucketLayout()
-        .equals(BucketLayoutProto.LEGACY)) {
+
+    // If the user has set a value, bucketInfo.hasBucketLayout() would be true,
```
Comment (rakeshadr) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketCreateRequest.java:155]: Please modify the comment to
```

    // bucketInfo.hasBucketLayout() would be true when user sets bucket layout.
    // Now, OM will create bucket with the user specified bucket layout.
    // When the value is not specified by the user, OM will use the
    // "ozone.default.bucket.layout" configured value.
```

**Review by rakeshadr — COMMENTED** [review]
@tanvipenumudy 
```
Error:  org.apache.hadoop.ozone.om.TestOMStartupWithBucketLayout.testRestartWithFSOLayout  Time elapsed: 11.725 s  <<< FAILURE!
java.lang.AssertionError: expected:<FILE_SYSTEM_OPTIMIZED> but was:<LEGACY>

Error:  org.apache.hadoop.ozone.om.TestObjectStoreWithFSO.testCreateBucketWithBucketLayout  Time elapsed: 0.052 s  <<< FAILURE!
java.lang.AssertionError: expected:<FILE_SYSTEM_OPTIMIZED> but was:<LEGACY>

TestDirectoryDeletingServiceWithFSO#test*
```

are failing because of the wrong default bucket layout values. Earlier LEGACY was replaced with OM.Server default. Now, with this patch an empty/null bucket layout (bucketInfo.hashasBucketLayout() == false) will use the OM.Server default bucket layout.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketCreateRequest.java (hunk 1 above)
Comment (tanvipenumudy) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketCreateRequest.java]: Thank you for the suggestions, I have incorporated the changes!

**Review by rakeshadr — APPROVED** [review]
+1 LGTM, thanks @tanvipenumudy for the contribution.

**Pull request merged**